src/lib.rs

```diff
@@ -160,3 +160,18 @@ mod platform_impl;
 pub mod window;
 
 pub mod platform;
+
+/// Wrapper for objects which winit will access on the main thread so they are effectively `Send`
+/// and `Sync`, since they always excute on a single thread.
+///
+/// # Safety
+///
+/// Winit can run only one event loop at the time and the event loop itself is tied to some thread.
+/// The objects could be send across the threads, but once passed to winit, they execute on the
+/// mean thread if the platform demands it. Thus marking such objects as `Send + Sync` is safe.
+#[doc(hidden)]
+#[derive(Clone, Debug)]
+pub struct SendSyncWrapper<T>(pub T);
```

Comment:
Not sure whether this object should be public at all, the only reason it's public is because we allow access to `WindowAttributes` and also building them with struct builder.

We could make it crate private and just provide a `get`/`set for `parent` on WindowAttributes?